### ISSUE TITLE
[doc] bug in Start Here!

### ISSUE THREAD

**Issue by christallire** [huggingface/accelerate#1422]
### System Info

```Shell
URL: https://huggingface.co/docs/accelerate/main/en/usage_guides/explore
```

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [ ] My own task or dataset (give details below)

### Reproduction

Go see the doc

### Expected behavior

There's a bug in the code of "Gradient Accumulation" 

```python
  from accelerate import Accelerator
  accelerator = Accelerator(
     gradient_accumulation_steps=2,
  )
  dataloader, model, optimizer scheduler = accelerator.prepare(
          dataloader, model, optimizer, scheduler
  )
  
  for batch in dataloader:
   with accelerator.accumulate(model):
        optimizer.zero_grad()
        inputs, targets = batch
        outputs = model(inputs)
        loss = loss_function(outputs, targets)
        accelerator.backward(loss)
        optimizer.step()
        scheduler.step()
````

`optimizer.zero_grad()` after the first line of `accumulate` clears the accumulated gradients when about to sync the gradient which prevents the model from training at all - it needs to be fixed. 

I've personally tried to fix myself but it's iframed in gradio, so cannot. wonder why would we need that anyways.

so It needs to be fixed either:
1) move `optimizer.zero_grad()` in the doc and all iframed examples
2) move `optimizer.zero_grad()` in the example and diff it to prevent readers from mistakenly not changing it like we did

@muellerzr

**Comment by muellerzr** [comment]
Thanks @christallire, fixed this now! In the future, the reason for the iframe is it's actually a hosted Gradio demo, shown here on spaces: https://huggingface.co/spaces/muellerzr/accelerate_examples/tree/main
